src/cache/cache-localstorage.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  CacheKey,
+  findExistingCacheKey,
+  ICache,
+  CACHE_KEY_PREFIX
+} from './shared';
+
+export class LocalStorageCache implements ICache {
+  public set(key: string, entry: unknown): Promise<void> {
+    localStorage.setItem(key, JSON.stringify(entry));
+    return Promise.resolve();
+  }
+
+  public get(key: string): Promise<unknown> {
+    return new Promise(resolve => {
+      const cacheKey = CacheKey.fromKey(key);
+      const payload = this.readJson(cacheKey);
+
+      if (!payload) resolve(null);
+
+      resolve(payload);
+    });
+  }
+
+  public remove(key: string): Promise<void> {
+    localStorage.removeItem(key);
+    return Promise.resolve();
+  }
+
+  public clear(): Promise<void> {
+    return new Promise(resolve => {
+      for (var i = localStorage.length - 1; i >= 0; i--) {
+        if (localStorage.key(i).startsWith(CACHE_KEY_PREFIX)) {
+          localStorage.removeItem(localStorage.key(i));
+        }
+      }
+
+      resolve();
+    });
+  }
```

Comment:
What is the reason that we use two different patterns here?

`get` and `clear` use `new Promise(resolve => ...)` while `set` and `remove` use `Promise.resolve()`.

Reply:
I used the former for longer blocks that could potentially return multiple times, and the latter for really short, simple blocks. Either way the result is the same.

Reply:
I ended up changing this and settling on using `Promise.resolve` everywhere for consistency 👍🏻